src/UnitfulRecipes.jl

```diff
@@ -63,6 +63,9 @@ end
 @recipe function f(f::Function, x::T) where T <: AVec{<:Union{Missing,<:Quantity}}
     x, f.(x)
 end
+@recipe function f(f::Function, u::Units)
+    x->f(x*u)
```

Comment:
To improve error messages, maybe replace this closure by a callable struct
```julia
struct WithUnit{F, U}
    f::F
    unit::U
end
(o::WithUnit)(x) = o.f(x*o.unit)
```

Reply:
Also I think you can do things like `heatmap(f)` for functions taking two inputs?

Reply:
I would have expected something like that too, but `f(x,y)=x+y;heatmap(f)` and `f(x,y)=x+y),surface(f)` error out.

The way I've implemented it now, this functionality should come automatically if this is implemented in Plots. Very difficult to test though.